build-util/src/bindgen.rs

```diff
@@ -79,6 +80,42 @@ pub fn generate(
     }
 }
 
+#[derive(Debug)]
+struct Callbacks;
+
+impl ParseCallbacks for Callbacks {
+    fn process_comment(&self, comment: &str) -> Option<String> {
+        let comment = comment.replace("@param[int]", "@param[in]");
+        let comment = comment.replace("[inout]", "[in,out]");
+
+        // doxygen_rs doesn't handle tabs well
+        let comment = comment.replace('\t', " ");
+        // Fixes links to functions and types
+        let comment = comment.replace("@see ::", "@see crate::");
+
+        // Escape square brackets for non-links
+        let comment =
+            lazy_regex::regex_replace_all!("\\[([\\d]+)\\]", &comment, |_, num: &str| format!(
+                "\\[{num}\\]"
+            ));
```

Comment:
I'd recommend using raw literals here, such as:
```rust
let comment =
    lazy_regex::regex_replace_all!(r"\[([\d]+)\]", &comment, |_, num: &str| dbg!(
        format!(r"\[{num}\]")
    ));
```
But I realized that this isn't working (you can check `act_exp_time` on `_sceNpDrmCheckActData`. I think that doxygen-rs is interpreting `\[0\]` as an unknown command and just ignoring it... Then I though I could escape the backslashes, but it doesn't work (you can even find `\` in the list of unsupported commands [here](https://github.com/Techie-Pi/doxygen-rs/issues/1)).

I thought that a simple solution would be to just move it after the doxygen transform, right? Turns out it doesn't seem like it works either........... As bindgen seems to be escaping the backslash, so `\[0\]` becomes `\\[0\\]` in the doc comment and the it ends up getting rendered. I seems like this is caused by the comment being included as a proc-macro2 Literal token (by using `quote`) which automatically escapes backslashes using char::escape_default. Sources: [0](https://github.com/rust-lang/rust-bindgen/blob/b5a6813c9a6feaeda38d7c672a1ed155f9e27266/bindgen/codegen/mod.rs#L1445) [1](https://github.com/rust-lang/rust-bindgen/blob/b5a6813c9a6feaeda38d7c672a1ed155f9e27266/bindgen/codegen/helpers.rs#L59) [2](https://github.com/dtolnay/quote/blob/bdb4b594076d78127b99a3da768e369499e324de/src/to_tokens.rs#L117-L117) [3](https://github.com/dtolnay/proc-macro2/blob/e94f4326bdd5a94e16be6da362af2e31bea52583/src/fallback.rs#L1027)

It could be nice to open an issue in bindgen, but at this point it we should probably workaround by doing something like adding an extra space? e.g.
```rust
let comment =
    lazy_regex::regex_replace_all!(r"\[([\d]+)\]", &comment, |_, num: &str| dbg!(
        format!(r"[ {num} ]")
    ));
```